server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java

```diff
@@ -294,8 +285,18 @@ public void readFromWithId(long id, StreamInput in) throws IOException {
         if (hasAggs = in.readBoolean()) {
             aggregations = InternalAggregations.readAggregations(in);
         }
-        pipelineAggregators = in.readNamedWriteableList(PipelineAggregator.class).stream().map(a -> (SiblingPipelineAggregator) a)
-                .collect(Collectors.toList());
+        //TODO update version after backport
+        if (in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
I think you need to check if `hasAggs` is true here because `aggregations` can be null.

Reply:
yes good catch